src/main/java/com/google/cloud/tools/managedcloudsdk/MessageListener.java

```diff
@@ -17,5 +17,18 @@
 package com.google.cloud.tools.managedcloudsdk;
 
 public interface MessageListener {
+  /**
+   * Process a full line. Implementers should add a newline to the end.
+   *
+   * @param line a full line (does not include newline character)
+   */
+  void messageLn(String line);
+
+  /**
+   * Process a raw message. Implementers should not add a newline to the end, it may contain newline
```

Comment:
I'm confused as to why we need two methods here, or what this interface is doing.

Reply:
A provider of messages can either send lines or raw messages. This is mostly for having to deal with messages like
```
processing . . . . . . . done.
```
so the output shows progress.

Reply:
I think I can change this though, and just have "message" and inject newlines on my own into this when reading line by line.